llvm/lib/ObjectYAML/ELFYAML.cpp

```diff
@@ -1599,8 +1599,11 @@ void MappingTraits<std::unique_ptr<ELFYAML::Chunk>>::mapping(
     // When the Type string does not have a "SHT_" prefix, we know it is not a
     // description of a regular ELF output section.
     TypeStr = getStringValue(IO, "Type");
-    if (TypeStr.starts_with("SHT_") || isInteger(TypeStr))
+    if (TypeStr.starts_with("SHT_") || isInteger(TypeStr)) {
       IO.mapRequired("Type", Type);
+      if (IO.error())
+        Type = ELF::SHT_NULL;
+    }
```

Comment:
what about the else branch here? won't we leave Type uninit as well?

Reply:
or do we know it's either `"Fill"` or `ELFYAML::SectionHeaderTable::TypeStr`, in which case we should add an `assert` for that

Reply:
On `!IO.error()` mapRequired will set `Type`, it's output arg.

Reply:
No, I mean else of 
```
if (TypeStr.starts_with("SHT_") || isInteger(TypeStr)) {
```

Reply:
Done